### ISSUE TITLE
Authenticate each post

### ISSUE THREAD

**Issue by ashefor** [Mayowa-Ojo/Newbie-backend-#2]
Right now a post made without a user logged in and one made with a user logged in are both the same and as such either post can be deleted by anybody.

**Comment by Mayowa-Ojo** [comment]
First off, a post can only be created if the user is logged in, because the create post route is protected (please confirm this).

**Comment by ashefor** [comment]
Yes it's protected obviously and the delete button too is protected unless user is logged in but right now, any logged in user can delete any post

**Comment by Mayowa-Ojo** [comment]
Okay, I'll authorize the delete route but on the client, you should also only display the delete button if the post was created by the current logged in user. Having a delete button there will make a user think they can delete a post. Bad UX

**Comment by ashefor** [comment]
The delete button only shows when a user is logged in and also if user created the post but right now there is no way to differentiate each post from others. You get? That's the challenge. Differentiating a post made by me from a post made by you because no userId  or something to differentiate

**Comment by Mayowa-Ojo** [comment]
Okay. Working on that

**Comment by Mayowa-Ojo** [comment]
On creating posts: The created post now has an author field. 

Slight changes: when you make a post request to create a post, of course, the user has to be logged in...you need to send the id of the current logged in user along with the request body in that meta field. There's now an **authorId** property in the meta field. Just set the id of the logged-in user to that authorId and then send the request.

- Pull and test this feature

**Comment by ashefor** [comment]
hmm will test and get back to you

**Comment by ashefor** [comment]
<img width="1440" alt="Screenshot 2019-10-08 at 12 33 16 PM" src="https://user-images.githubusercontent.com/31312830/66392588-35642f80-e9c8-11e9-9a98-91ed79534237.png">
is this not how a post data should look like? if yes why am I getting a "user not found" error?

**Comment by Mayowa-Ojo** [comment]
Okay. so all settled now.

For the record, the meta field should contain an `authorId` field pointing to the id of the current logged in user.
